ts/features/bonus/bpd/screens/details/transaction/BpdTransactionsScreen.tsx

```diff
@@ -27,29 +32,46 @@ const dataForFlatList = (
  * TODO: scroll to refresh, display error, display loading
  * @constructor
  */
-const BpdTransactionsScreen: React.FunctionComponent<Props> = props => (
-  <BaseScreenComponent goBack={true} headerTitle={I18n.t("bonus.bpd.title")}>
-    <SafeAreaView style={IOStyles.flex}>
-      <View style={[IOStyles.horizontalContentPadding, IOStyles.flex]}>
-        <View spacer={true} large={true} />
-        <H1>{I18n.t("bonus.bpd.details.transaction.title")}</H1>
-        <View spacer={true} />
-        <FlatList
-          data={dataForFlatList(props.transactionForSelectedPeriod)}
-          renderItem={transaction => (
-            <BpdTransactionItem transaction={transaction.item} />
+const BpdTransactionsScreen: React.FunctionComponent<Props> = props => {
+  const transactions = dataForFlatList(props.transactionForSelectedPeriod);
+
+  const trxSortByDate = [...transactions].sort((trx1, trx2) =>
+    compareDesc(trx1.trxDate, trx2.trxDate)
+  );
+
+  return (
+    <BaseScreenComponent goBack={true} headerTitle={I18n.t("bonus.bpd.title")}>
+      <SafeAreaView style={IOStyles.flex}>
+        <View style={[IOStyles.horizontalContentPadding, IOStyles.flex]}>
+          <View spacer={true} large={true} />
+          <H1>{I18n.t("bonus.bpd.details.transaction.title")}</H1>
+          <View spacer={true} />
+          {pot.isSome(props.selectedAmount) && props.selectedPeriod && (
```

Comment:
Using `format(trxSortByDate[0].trxDate` we assume that array will be always not empty.
This is a strong assumption and could generate an error in specific conditions

```typescript
import { index } from "fp-ts/lib/Array";
const maybeFirstDate = index(0,trxSortByDate).map(i => i.trxDate);
...
{pot.isSome(props.selectedAmount) && props.selectedPeriod && maybeFirstDate.isSome()
```

Reply:
You're right, I made an additional fix.
Using this three conditions together (`pot.isSome(props.selectedAmount) && props.selectedPeriod && maybeFirstDate.isSome()`) may cause to not show the summary component even if it should be displayed.

For example: The selected period and amount are correctly available, but no transactions is still available, the condition above would hide all the text block.

I think this solution may cover that case easier:
```javascript
  const lastUpdateDate = format(
    index(0, trxSortByDate).fold(new Date(), t => t.trxDate),
    "DD MMMM YYYY"
  );
```

Reply:
@CrisTofani for me, it's ok.
So the fallback, if the date array is empty, will be `now`.
Do you confirm?

Reply:
yes it is!